stdlib/Distributed/src/cluster.jl

```diff
@@ -1310,14 +1311,17 @@ function process_opts(opts)
         end
     end
 
+    # Propagate --threads to workers
+    exeflags = opts.nthreads > 0 ? `--threads=$(opts.nthreads)` : ``
```

Comment:
Is it intentional that this does not look at `Threads.nthreads()`?  That is to say, `JULIA_NUM_THREADS` does not propagate to the remote workers but `--threads` does?

Reply:
My thinking was that if you use the env variable, then you probably use the env variable on the workers too. This is just a way to override it.

Reply:
> you probably use the env variable on the workers too

I think it's useful and easy for "local" workers. But isn't this hard to do for "remote" workers (e.g., launched via ssh)? IIUC, there is no easy way to use environment variables in remote workers other than creating a "shim" script and setting the environment variable there.

I just thought it may be confusing that `Threads.nthreads()` does not propagate only when the workers are in different machines _and_ it is set via `JULIA_NUM_THREADS` in the main process.

I don't have a strong opinion on what is right default here, though.